fabtests/regression/shm_sighandler_test.c

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2022 Intel Corporation.  All rights reserved.
+ *
+ * This software is available to you under a choice of one of two
+ * licenses.  You may choose to be licensed under the terms of the GNU
+ * General Public License (GPL) Version 2, available from the file
+ * COPYING in the main directory of this source tree, or the
+ * BSD license below:
+ *
+ *     Redistribution and use in source and binary forms, with or
+ *     without modification, are permitted provided that the following
+ *     conditions are met:
+ *
+ *      - Redistributions of source code must retain the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer.
+ *
+ *      - Redistributions in binary form must reproduce the above
+ *        copyright notice, this list of conditions and the following
+ *        disclaimer in the documentation and/or other materials
+ *        provided with the distribution.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+/*
+ * Regression test for https://github.com/ofiwg/libfabric/pull/7605:
+ * "prov/shm: Properly chain the original signal handlers".
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <sys/wait.h>
+#include <rdma/fabric.h>
+
+int main(void)
+{
+	int child;
+	int status;
+
+	if ((child = fork())) {
+		usleep(100000); /* give child time to finish initialization */
+		kill(child, SIGINT);
+		usleep(100000); /* give child time to handle the signal */
+		kill(child, SIGKILL);
+		waitpid(child, &status, 0);
+		if (WTERMSIG(status) == SIGINT) {
+			printf("pass\n");
+			exit(0);
+		}
+		printf("fail\n");
+		exit(1);	
+	} else {
+		struct fi_info *fi;
+
+		if (fi_getinfo(FI_VERSION(1,0), NULL, NULL, 0, NULL, &fi))
+			fi_freeinfo(fi);
```

Comment:
Does shm replace the signal handlers when fi_getinfo is called?

Reply:
It happens in the provider initialization code (referred by the macro SHM_INI).

Reply:
@aingerson - Why is smr_reg_sig_handler() called from SHM_INI?  Couldn't this be deferred until an EP is created?

Reply:
It could be deferred to the EP. We would just need to add some code to make sure it's only called once but that's easy enough. Would you like me to do that? We'll need to change this test to match

Reply:
I think it's worth changing as that would limit the impact to when only the shm provider is actually used, rather than simply being present.  The test would probably only need to add a call to ft_init_fabric().  The test changes could go in before shm was updated.  And the test could then verify that shm was present in the library as part of its check.